content/posts/faster-rfft-python/index.md

```diff
@@ -0,0 +1,284 @@
++++ 
+draft = true
+date = 2025-11-25T01:24:53-05:00
+title = "Getting Faster RFFT in Python"
+description = ""
+slug = ""
+authors = []
+tags = []
+categories = []
+externalLink = ""
+series = []
++++
+
+
+Real Fast Fourier Transform (RFFT) is an algorithm that converts a real-valued signal from its time domain into its frequency components. Different libraries in python, such as NumPy, SciPy, and [pyFFTW](https://pyfftw.readthedocs.io/en/latest/source/pyfftw/builders/builders.html), provide an API for RFFT. However, their performance can vary. The focus of this post is to demonstrate how to use pyFFTW properly. pyFFTW is a Python wrapper for the [FFTW](https://www.fftw.org), which is known for its speed and efficiency in computing Fourier transforms. 
+
+
+## Installation
+[FFTW](https://www.fftw.org) is one of the requirements of pyFFTW that needs to be installed first, and it does not come bundled with pyFFTW. The easiest way to install it is through `conda`. I assume you have `conda` installed, and you know how to create and activate a conda environment. The minimum python version for pyFFTW is `3.11` (as of November 2025).
+
+Once you are in your environment, install `FFTW` by running the following command:
+```bash
+conda install conda-forge::fftw
+```
+
+And, the other dependencies can be installed when installing pyFFTW:
+
+```bash
+conda install conda-forge::pyfftw
+``` 
+
+
+Now, we are ready to use pyFFTW in our code!
+
+
+## Example Code
+The following code demonstrates how to use pyFFTW to perform RFFT on a real-valued signal. 
+
+```python
+import numpy as np
+import pyfftw
+
+T = np.random.rand(4)
+rfft_obj = pyfftw.builders.rfft(np.empty(len(T), dtype='float64'))
+R = rfft_obj(T)
```

Comment:
Get rid of anything that comes after this line. Our goal is to simply show how one can compute the rfft of `T`